### PR TITLE
Enhance install process by generating config files from templates

### PR DESCRIPTION
**Pull request by JocelynDelalande**
Motivations are:

- https://github.com/spiral-project/ihatemoney/issues/195#issuecomment-291607821
- frustration (re)writing installation and upgrade docs

Writing docs is a good exercise : if something is too much complicated to explain, maybe it is simply too much complicated.

So this is a simplification attempt of the install process.

This may be extended in the future (the script may ask questions or receive arguments), but that is out of the scope of this PR.

This is (IMHO) a requirement for #274 (put another, I ended up with this PR trying to write the PR for #274 :-p)

What do you think ?

### CODE REVIEW

**Comment by almet** [comment]
Great, I think it's better this way too. One small adjustment might be to change the command name to `generate-config` as I tend to prefer verbs in the command actions.

Also, some people might prefer to copy/paste the settings files (even with the Jinja2 variable inside them), it might be good to point their location in the docs as well?

**Comment by JocelynDelalande** [comment]
Thanks for fast answer !

> Great, I think it's better this way too. One small adjustment might be to change the command name to generate-config as I tend to prefer verbs in the command actions.

Done

> Also, some people might prefer to copy/paste the settings files (even with the Jinja2 variable inside them), it might be good to point their location in the docs as well?

I tend to think it bloats the documentation, but well… DONE also :-). I added a note with a link to the github folder.

File: ihatemoney/conf-templates/ihatemoney.cfg.j2 [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# Enter an absolute path here.
+SQLALCHEMY_DATABASE_URI = 'sqlite:///var/lib/ihatemoney/ihatemoney.sqlite'
+SQLACHEMY_ECHO = DEBUG
+
+# Will likely become the default value in flask-sqlalchemy >=3 ; could be removed
+# then:
+SQLALCHEMY_TRACK_MODIFICATIONS = False
+
+# This secret key is random and auto-generated, it protects cookies and user sessions
+SECRET_KEY = "{{ secret_key }}"
```
Comment (almet) [inline ihatemoney/conf-templates/ihatemoney.cfg.j2]: I especially like this as it's now possible to generate a secret key without even the user noticing it :)

File: ihatemoney/default_settings.py [hunk 2]
```diff
@@ -1,37 +1,10 @@
...
-# documentation, available online at
-# http://ihatemoney.readthedocs.io/en/latest/installation.html#configuration
-
-# Turn this on if you want to have more output on what's happening under the
-# hood.
-DEBUG = False
-
-# The database URI, reprensenting the type of database and how to connect to it.
-# Enter an absolute path here.
+# Verbose and documented settings are in conf-templates/ihatemoney.cfg.j2
```
Comment (almet) [inline ihatemoney/default_settings.py]: Thanks for not duplicating this :)

**Pull request merged**